https://github.com/shadow-maint/shadow/pull/463

From: Sam James <sam@gentoo.org>
Date: Mon, 20 Dec 2021 01:24:16 +0000
Subject: [PATCH] libsubid: fix defining SONAME version

We were overriding this when --enable-shared was passed. We can actually
just dump the conditional logic as libtool will do the right thing for
us here anyway.

Without this patch, libsubid is installed as .0.

Signed-off-by: Sam James <sam@gentoo.org>
--- a/libsubid/Makefile.am
+++ b/libsubid/Makefile.am
@@ -1,10 +1,6 @@
 lib_LTLIBRARIES = libsubid.la
-if ENABLE_SHARED
-libsubid_la_LDFLAGS = -Wl,-soname,libsubid.so.@LIBSUBID_ABI@ \
-	-shared -version-info @LIBSUBID_ABI_MAJOR@
-endif
 libsubid_la_SOURCES = api.c
-libsubid_la_LDFLAGS = -export-symbols-regex '^subid_'
+libsubid_la_LDFLAGS = -version-info @LIBSUBID_ABI_MAJOR@ -export-symbols-regex '^subid_'
 
 pkginclude_HEADERS = subid.h
 
